include/bout/mesh.hxx

```diff
@@ -708,7 +727,7 @@ class Mesh {
   
   GridDataSource *source; ///< Source for grid data
   
-  std::map<CELL_LOC, std::shared_ptr<Coordinates> > coords_map; ///< Coordinate systems at different CELL_LOCs
+  std::map<CELL_LOC, std::unique_ptr<Coordinates> > coords_map; ///< Coordinate systems at different CELL_LOCs
```

Comment:
I'm not sure about changing this to `unique_ptr`. I know we currently hand out a raw pointer, but that should change, and returning a `unique_ptr` says "I'm giving ownership over to you", whereas a `shared_ptr` says "You can have a look at this, but I will keep hold of it too"